From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@niif.hu>
Date: Mon, 21 Dec 2015 18:04:02 +0100
Subject: Libraries go after linker flags

Ensure that options like -Wl,--as-needed are effective for the stack
libraries.
---
 lib/cluster/Makefile.am |    4 ++--
 mcp/Makefile.am         |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/cluster/Makefile.am b/lib/cluster/Makefile.am
index 29daeb2..c063340 100644
--- a/lib/cluster/Makefile.am
+++ b/lib/cluster/Makefile.am
@@ -28,8 +28,8 @@ header_HEADERS =
 lib_LTLIBRARIES	= libcrmcluster.la 
 
 libcrmcluster_la_SOURCES = election.c cluster.c membership.c
-libcrmcluster_la_LDFLAGS = -version-info 5:0:1 $(CLUSTERLIBS) 
-libcrmcluster_la_LIBADD  = $(top_builddir)/lib/common/libcrmcommon.la $(top_builddir)/lib/fencing/libstonithd.la
+libcrmcluster_la_LDFLAGS = -version-info 5:0:1
+libcrmcluster_la_LIBADD  = $(top_builddir)/lib/common/libcrmcommon.la $(top_builddir)/lib/fencing/libstonithd.la $(CLUSTERLIBS)
 libcrmcluster_la_DEPENDENCIES = $(top_builddir)/lib/common/libcrmcommon.la $(top_builddir)/lib/fencing/libstonithd.la 
 
 if BUILD_CS_SUPPORT
diff --git a/mcp/Makefile.am b/mcp/Makefile.am
index eb579cc..c3caf19 100644
--- a/mcp/Makefile.am
+++ b/mcp/Makefile.am
@@ -33,7 +33,7 @@ noinst_HEADERS		= pacemaker.h
 
 pacemakerd_SOURCES	= pacemaker.c corosync.c 
 pacemakerd_LDADD	= $(top_builddir)/lib/cluster/libcrmcluster.la $(top_builddir)/lib/common/libcrmcommon.la
-pacemakerd_LDFLAGS	= $(CLUSTERLIBS)
+pacemakerd_LDADD	+= $(CLUSTERLIBS)
 
 clean-generic:
 	rm -f *.log *.debug *.xml *~
